src/Layers/FeatureLayer/FeatureManager.js

```diff
@@ -383,22 +383,41 @@
       return this._service.query();
     },
 
+    _getMetadata: function(callback, context){
+      if(this._metadata){
+        callback(undefined, this._metadata, this);
+      } else {
+        this.metadata(L.Util.bind(function(error, response, context) {
+          this._metadata = response;
+          callback(error, this._metadata, this);
+        }, this));
+      }
+    },
+
     addFeature: function(feature, callback, context){
-      this._service.addFeature(feature, function(error, response){
-        if(!error){
-          this.refresh();
-        }
-        if(callback){
-          callback.call(context, error, response);
-        }
+      this._getMetadata(function(error, metadata, context){
+        context._service.addFeature(feature, function(error, response){
+          if(!error){
+            // assign ID from result to appropriate objectid field from service metadata
+            feature.properties[metadata.objectIdField] = response.objectId;
+
+            // we also need to update the geojson id for createLayers() to function
+            feature.id = response.objectId;
+            context.createLayers([feature]);
+          }
+          if(callback){
+            callback.call(context, error, response);
+          }
+        }, context);
```

Comment:
With the above changes to the `_getMetadata()` method you should pass `this` here not context.